include/deal.II/base/index_set.h

```diff
@@ -344,6 +344,14 @@ class IndexSet
   IndexSet
   get_view(const size_type begin, const size_type end) const;
 
+  /**
+   * Split the set of DoFs (typically locally owned or relevant) in @p whole_set into blocks
+   * given by the @p dofs_per_block structure.
+   */
+  static std::vector<IndexSet>
+  split_by_block(const std::vector<types::global_dof_index> &dofs_per_block,
+                 const IndexSet &                            whole_set);
```

Comment:
If you declare it as a member function, I don't see a reason to make it `static`. Just use the instance and drop the `IndexSet` argument.

Reply:
I tried to follow your idea, but maybe I did not understand it right. Now  I get again the error that I "call a non-static member function without an object argument"

Reply:
Use
```
whole_set.split_by_block(dofs_per_block);
```
instead of
```
IndexSet::split_by_block(dofs_per_block, whole_set);
```

Reply:
okay, I got it